docs/operate/get-started/other-hardware/manage-modules.md

```diff
@@ -21,119 +20,61 @@ date: "2024-06-30"
-   For example, if you have altered your model's description, or adjusted the endpoint name, you'll need to update `meta.json` with these changes.
-
-1. For modules written in Python, you should package your module files as an archive first, before uploading.
-   Other languages can proceed to the next step to upload their module directly.
-   To package a module written in Python, run the following command from the same directory as your `meta.json` file:
-
-   {{< tabs >}}
-   {{% tab name="Packaged executable" %}}
-
-   ```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
-   tar -czf module.tar.gz <PATH-TO-EXECUTABLE>
-   ```
-
-   where `<PATH-TO-EXECUTABLE>` is the [packaged executable](/operate/get-started/other-hardware/#test-your-module-locally).
-
-   {{% /tab %}}
-   {{% tab name="Using venv" %}}
+Use GitHub Actions to automatically build and deploy your new module version when you create a tag or release in GitHub:
 
-   ```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
-   tar -czf module.tar.gz run.sh requirements.txt src
-   ```
-
-   Where `run.sh` is your [executable file](/operate/get-started/other-hardware/#test-your-module-locally), `requirements.txt` is your [pip dependency list file](/operate/get-started/other-hardware/#test-your-module-locally), and `src` is the directory that contains the source code of your module.
+1. Edit your module code and update the [`meta.json`](/operate/get-started/other-hardware/#metajson-reference) file if needed.
 
-   {{% /tab %}}
-   {{< /tabs >}}
-
-   Supply the path to the resulting archive file in the next step.
+   {{% alert title="Note" color="note" %}}
 
-1. Run `viam module upload` to upload your custom module to the Viam Registry:
+   If you used `viam module generate` to create your module and enabled cloud build, all you need to do to trigger a new build is create a tag and publish a release in GitHub as you did when you [first published the module](/operate/get-started/other-hardware/#upload-your-module).
 
-   ```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
-   viam module upload --version <version> --platform <platform> <module-path>
-   ```
+   {{% /alert %}}
 
-   For example, the following command uploads a module compressed as an archive named `my-module.tar.gz` to the Viam Registry, and increments the [`version`](/dev/tools/cli/#using-the---version-argument) of the module to version `1.0.1`:
+1. If you did not use the Viam CLI generator and enable cloud build, you can set up one of the following GitHub actions up manually:
 
-   ```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
-   viam module upload --version 1.0.1 --platform darwin/arm64 my-module.tar.gz
+   {{< tabs >}}
+   {{% tab name="build-action (Recommended)" %}}
+
+   The `build-action` GitHub action provides a simple cross-platform build setup for multiple platforms: x86, ARM Linux, and macOS.
+
+   Add this to your GitHub workflow:
```

Comment:
Unless people already know how GitHub Actiosn work, I think we need to link out to GH docs here on this. Or explain how to do this. I think we're also assuming here that they have already pushed to GitHub - that probably also needs clarifying

Reply:
Okay making some tweaks. I must say the GitHub UI is pretty intuitive--I hadn't gone through this workflow before recently and I found it easy to follow without too much documentation. I know I'm just a sample size of 1 but I want to keep this concise and we do have the link to more docs below, and i've added the link above too. LMK what you think of latest commit.

Reply:
ok, could add it as a prerequisite but optional

Reply:
For adding the workflow, they'd generally be creating a new file for the build action, I think. Either way this currently still requires you to know where workflows are and how to add them, easier to specify:
```suggestion
   In your repository, create a workflow file called <FILE>build.yml</FILE> in the <FILE>.github/workflows<FILE> directory:
```